cpp/src/parquet/column_reader.cc

```diff
@@ -1028,12 +1030,17 @@ class TypedColumnReaderImpl : public TypedColumnReader<DType>,
   // and number of values to read. This function is called before reading values.
   void ReadLevels(int64_t batch_size, int16_t* def_levels, int16_t* rep_levels,
                   int64_t* num_def_levels, int64_t* values_to_read) {
```

Comment:
Is it useful to have a  `int64_t* num_def_levels` if it's always going to be equal to `batch_size`? Can we remove this argument?

Reply:
```
batch_size = std::min(batch_size, this->available_values_current_page());
```

The `batch_size` would be adjusted by the code here.